### PR TITLE
Create game

### LINKED ISSUE #3: API Endpoint: Create Game
```
As a user
  when I send a POST request to /api/v1/games
  and I include a JSON body with the username I wish to use during the game
I receive a JSON response including
  a unique game code that I can share for other players to join the game
  a channel name I can subscribe to for game events
  my approved username
  an access token I can use to communicate with the server about this game
```

#### Request
```http
POST /api/v1/games
```
```js
{
  "name": "<desired_username>" // e.g. "Archer"
}
```

#### Successful Response
```http
HTTP/1.1 201 Created
```
```js
{
  "game_channel": "game_x", // e.g. "game_9aZReVkGAotVahGLS88vEnYw"
  "invite_code": "L6J5suAqTsKUAjEXm5swoEUN", // a unique token to be shared with friends
  "name": "Archer",
  "token": "uuxHQc7djqQuzWgJxAp5r1vy" // a unique token to identify the player
}
```

#### Failed Response
```http
HTTP/1.1 401 Unauthorized
```
```js
{
  "error": "You must provide a username"
}
```
The request may fail if the username is not provided.

### PR DESCRIPTION
**Pull request by joequincy**
# Description
Closes #3

```
As a user
  when I send a POST request to /api/v1/games
  and I include a JSON body with the username I wish to use during the game
I receive a JSON response including
  a unique invite code that I can share for other players to join the game
  a channel name I can subscribe to for game events
  my approved username
  an access token I can use to communicate with the server about this game
```

In order to begin a game, the first player must first request that the server create a new game room. To do this, the UI will send a POST request as defined in the user story above. The server will then create a User in its database*, create a new Game, and attach the user to the game as a Player. During this creation process, the server will generate unique tokens for accessing the game's primary Websockets channel and its "intel" channel, for inviting friends to the game, and for uniquely identifying the current player.

It will return the invite code, the channel name, and the player's access token, as well as confirming the requested name for the user.

*_At this time, we are not establishing accounts, so when the users end their game, we will need to make sure we delete all connected users from the database._

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:

## Has This Been Tested?

- [ ] No
- [x] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
Additional validation tests for using secure tokens have been added.
Added a test verifying the successful response for creating a game has the correct status and its body has the correct format.
Added a test verifying that if the end user does not submit a username with the request, it will fail with an error message explaining the issue.
- [x] Have any prior tests been changed?
If so, please explain:
The Player instance will be created before teams and roles have been assigned, so validations ensuring the presence of these attributes have been removed.

# Additional notes:
The Schema has been modified to support the addition of the various token attributes. The diagram image has been updated. When integrating these changes into your local branch(es), the database will need to be recreated with `rails db:{drop,create,migrate}`.

# Checklist:

- [x] My code has no unused/commented out code
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: db/seeds/cards.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
...
+
+words.each do |word|
+  Card.create!(word: word)
```
Comment (rdren0) [inline db/seeds/cards.rb]: Thanks for using the word list I provided

File: spec/models/game_spec.rb [hunk 2]
```diff
@@ -1,6 +1,12 @@
 require 'rails_helper'
 
 RSpec.describe Game, type: :model do
+  describe "Validations" do
+    it { should have_secure_token(:game_key) }
+    it { should have_secure_token(:intel_key) }
+    it { should have_secure_token(:invite_code) }
```
Comment (rdren0) [inline spec/models/game_spec.rb]: How does the game key and invite code differ? Could they be the same?

File: app/models/game.rb [hunk 3]
```diff
@@ -7,4 +7,8 @@ class Game < ApplicationRecord
 
   has_many :hints
   has_many :guesses
+
+  has_secure_token :game_key
```
Comment (rdren0) [inline app/models/game.rb]: Thanks for incorporating Auth Tokens

File: spec/models/game_spec.rb (hunk 2 above)
Comment (joequincy) [reply spec/models/game_spec.rb]: My thought process is that if the invite code is the same as the key to connect to the game channel, anyone with that key could subscribe to the channel and get game data updates even if they're not actually part of the game. This was a late-night decision though, so there may be alternative ways of handling this I haven't considered. I'd be happy to talk it over first thing when we group up today!

**Pull request merged**